plugin/src/main/java/org/opensearch/ml/processor/MLInferenceIngestProcessor.java

```diff
@@ -0,0 +1,253 @@
+    public static final String DEFAULT_OUTPUT_FIELD_NAME = "inference_results";
+
+    protected MLInferenceIngestProcessor(
+        String model_id,
+        List<Map<String, String>> input_map,
+        List<Map<String, String>> output_map,
+        Map<String, String> model_config,
+        String tag,
+        String description,
+        boolean ignoreMissing,
+        ScriptService scriptService,
+        Client client
+    ) {
+        super(tag, description);
+        this.mlModelUtil = new MLModelUtil(model_id, input_map, output_map, model_config);
+        this.ignoreMissing = ignoreMissing;
+        this.scriptService = scriptService;
+        this.client = client;
+    }
+
+    /**
+     * overwrite in this execute method,
+     * when batch inference is available,
+     * to support async multiple predictions.
+     */
+    @Override
+    public void execute(IngestDocument ingestDocument, BiConsumer<IngestDocument, Exception> handler) {
+
+        List<Map<String, String>> process_input = mlModelUtil.getInput_map();
+        List<Map<String, String>> process_output = mlModelUtil.getOutput_map();
+        int i = 0;
+        int round = (process_input != null) ? process_input.size() : 0;
+
+        process_predictions(ingestDocument, handler, process_input, process_output, i, round);
+
+    }
+
+    @Override
+    public IngestDocument execute(IngestDocument ingestDocument) throws Exception {
+        return ingestDocument;
+    }
+
+    /**
+     * process predictions for one model for multiple rounds of predictions
+     * ingest documents after prediction rounds are completed,
+     * when no input mappings provided, default to add all fields to model input fields,
+     * when no output mapping provided, default to output as
+     * "inference_results" field (the same format as predict API)
+     */
+    private void process_predictions(
+        IngestDocument ingestDocument,
+        BiConsumer<IngestDocument, Exception> handler,
+        List<Map<String, String>> process_input,
+        List<Map<String, String>> process_output,
+        int i,
+        int round
+    ) {
+        if (i >= round && i != 0) {
+            handler.accept(ingestDocument, null);
+        } else {
+            Map<String, String> modelParameters = new HashMap<>();
+            if (mlModelUtil.getModel_config() != null) {
+                modelParameters.putAll(mlModelUtil.getModel_config());
+            }
+            // when no input mapping is provided, default to read all fields from documents as model input
+            if (round == 0) {
+                Set<String> documentFields = ingestDocument.getSourceAndMetadata().keySet();
+                for (String field : documentFields) {
+                    getMappedModelInputFromDocuments(ingestDocument, modelParameters, field, field);
+                }
+
+            } else {
+                Map<String, String> inputMapping = process_input.get(i);
```

Comment:
Let's use camel case for java variables.

Reply:
chnaged